core/mempool/src/lib.rs

```diff
@@ -102,7 +114,16 @@ impl Pool {
             ));
         }
         self.known_to.write().expect(POISONED_LOCK_ERR).insert(transaction.get_hash(), HashSet::new());
-        self.transactions.write().expect(POISONED_LOCK_ERR).insert(transaction);
+        let mut guard = self.transactions.write().expect(POISONED_LOCK_ERR);
+        match guard.entry(transaction.body.get_originator()) {
```

Comment:
```
self.transactions.write().expect(POISONED_LOCK_ERR).entry(transaction.body.get_originator()).or_insert_with(BTreeMap::new).insert(transaction.body.get_nonce(), transaction);
```

Also, I am not sure the rational behind `BTreeMap` here.

Reply:
Fixed